src/handler/mod.rs

```diff
@@ -1241,6 +1256,28 @@ impl Handler {
                 }
             }
         }
+        // fail all active requests
+        for req in self
+            .active_requests
+            .remove_requests(node_address)
+            .unwrap_or_default()
+        {
+            match req.id() {
+                HandlerReqId::Internal(_) => {
+                    // Do not report failures on requests belonging to the handler.
+                }
+                HandlerReqId::External(id) => {
+                    if let Err(e) = self
+                        .service_send
+                        .send(HandlerOut::RequestFailed(id.clone(), error.clone()))
+                        .await
+                    {
+                        warn!("Failed to inform request failure {e}")
+                    }
+                }
+            }
+        }
+        self.remove_expected_response(node_address.socket_addr);
```

Comment:
the count of expected responses depends on the number of sent requests, so this needs to be moved up